test/hold_effect_eject_button.c

```diff
@@ -0,0 +1,184 @@
+#include "global.h"
+#include "test_battle.h"
+
+ASSUMPTIONS
+{
+    ASSUME(gItems[ITEM_EJECT_BUTTON].holdEffect == HOLD_EFFECT_EJECT_BUTTON);
+}
+
+SINGLE_BATTLE_TEST("Eject Button is not triggered when there is nothing to switch in")
+{
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET) { Item(ITEM_EJECT_BUTTON); }
+        PLAYER(SPECIES_WOBBUFFET) { HP(0); }
```

Comment:
Is there a specific reason the player has 0 HP here?

Reply:
So the AI isn't switching a dead mon to battle.